### PR TITLE
Fixes regional endpoints (#23)

### PR DESCRIPTION
**Pull request by ktonon**

### CODE REVIEW

File: src/AWS/Config.elm [hunk 1]
```diff
@@ -8,11 +8,27 @@ type alias Credentials =
...
+
 type alias Service =
     { serviceName : String
+    , endpoint : Endpoint
     , version : String
-    , xAmzJsonVersion : String
+    , xAmzJsonVersion : Maybe String
     , xAmzTargetPrefix : String
-    , host : String
-    , region : String
```
Comment (ktonon) [inline src/AWS/Config.elm:17]: Don't need region if the service is not regional (such as IAM)

File: src/AWS/Config.elm [hunk 2]
```diff
@@ -8,11 +8,27 @@ type alias Credentials =
...
+    String
+
+
+type alias Host =
+    String
+
+
+type Endpoint
+    = RegionalEndpoint EndpointPrefix Region
+    | GlobalEndpoint Host
```
Comment (ktonon) [inline src/AWS/Config.elm:25]: Region is wrapped up in `RegionalEndpoint`, and `GlobalEndpoint` doesn't need it

File: src/AWS/Http.elm [hunk 3]
```diff
@@ -34,9 +35,22 @@ unsignedRequest target method uri params decoder =
...
+
+
+host : Endpoint -> String
+host endpoint =
+    case endpoint of
+        RegionalEndpoint prefix region ->
+            prefix ++ "." ++ region ++ ".amazonaws.com"
+
+        GlobalEndpoint host ->
+            host
```
Comment (ktonon) [inline src/AWS/Http.elm:53]: Map and endpoint to a hostname

File: src/AWS/Signers/V4.elm [hunk 4]
```diff
@@ -126,11 +126,22 @@ signature creds config date toSign =
...
 
+regionForAuth : AWS.Config.Endpoint -> String
+regionForAuth endpoint =
+    case endpoint of
+        AWS.Config.RegionalEndpoint _ region ->
+            region
+
+        AWS.Config.GlobalEndpoint _ ->
+            -- See http://docs.aws.amazon.com/general/latest/gr/sigv4_changes.html
+            "us-east-1"
```
Comment (ktonon) [inline src/AWS/Signers/V4.elm:142]: Docs say to use `us-east-1` for global endpoints for the purpose of v4 auth

File: src/AWS/Signers/V4.elm [hunk 5]
```diff
@@ -143,4 +154,11 @@ stringToSign algorithm date scope canon =
 
 jsonContentType : AWS.Config.Service -> String
 jsonContentType config =
-    "application/x-amz-json-" ++ config.xAmzJsonVersion ++ "; charset=utf-8"
+    (case config.xAmzJsonVersion of
+        Just version ->
+            "application/x-amz-json-" ++ version
+
+        Nothing ->
+            "application/json"
```
Comment (ktonon) [inline src/AWS/Signers/V4.elm:162]: Some apis do not specify a json version, so we'll just use vanilla `application/json` in that case. Haven't verified this in the AWS docs yet.

**Review by nickmonad — APPROVED** [review]
lgtm

**Pull request merged**